src/dqe_idx.erl

```diff
@@ -227,6 +228,28 @@ metrics(Collection) ->
 
 %%--------------------------------------------------------------------
 %% @doc
+%% Returns a list of metric path suffixes of `Depth' that are prefixed
+%% by the given probe `Prefix', which can also be empty.
+%%
+%% For example:
+%% metrics(<<"collection">>, <<>>, 1) -> [<<"base">>].
+%% metrics(<<"collection">>, <<"base">>, 1) -> [<<"cpu">>].
+%% metrics(<<"collection">>, <<>>, 2) -> [[<<"base">>.<<"cpu">>]].
+%%
+%% Actual results are binary-encoded in `dproto' encoding.
+%% @end
+%%--------------------------------------------------------------------
+-spec metrics(Collection::collection(),
+              Prefix::metric(),
+              Depth::pos_integer()) ->
+                    {ok, [Metric::metric()]} |
```

Comment:
I think they are different interfaces, data from the idnexer seesm to be usually used in list form, for both reading and writing. It only gets converted to binary when accesing the (second interface) dalmateirndb/dproto.

that the binary representation is used in FE was a simple thing of 'convineince' it happend to be there so ti was easy to use that as serialization instead of making something more URL appropriate.

The contrac in both forms will still be enforced using dialyzer, as a matter of fact `[binary()]` is a better and easyer to enforce contract then `binary()`. The simple `binary()` contract quietly assumes that the binary is conform to something (we can not check against at compile/dialyzer time!) while `[binary()]` does not impose any special form on the part of the binary(), and dialyzer can verify weather the data is a list with binaries or not. The example `[<<"base">>, [[[<<"cpu">>]]]]` would not pass those tests!